### PR TITLE
Add support for MySQL/MariaDB and Postgres + fixes

### PR DESCRIPTION
**Pull request by setrofim**
Add support for "mysql" and "pgx" drivers (for MySQL/MariaDB and
Postgres, respectively). This required the following changes to the
"sql" kvstore backend:

- "key" column had to be renamed to "kv_key", as the former is a
  reserved word in MySQL. "vals" column has also been renamed to
  "kv_val" for consistency (also, the old name implied that the column
  contained an array of values, where as it is a single string value).
- Queries are now assembled using squirrel, which allows switching the
  placeholder symbol. This is necessary because MySQL and Postgres
  expect different syntax for placeholders.

Documentation for KVStore has been amended to reflect that only three
drivers can be enabled via configuration for the services, and the
described support for other drivers is in the context of other code
using the KVStore implementation. (Plus minor line length
re-formatting.)

Native deployment has been augmented with optional config/init scripts
for the other SQL drivers, and documentation on how to enable them.

Additionally:

- Update end-to-end input corims to the new format
- Allow -f and -s flags at the same time when creating a native deployment

### CODE REVIEW

File: kvstore/sql.go [hunk 1]
```diff
@@ -109,18 +123,24 @@ func (o SQL) Get(key string) ([]string, error) {
...
 
-	// nolint:gosec
-	// o.TableName has been checked by isSafeTblName on init
-	q := fmt.Sprintf("SELECT DISTINCT vals FROM %s WHERE key = ?", o.TableName)
+	query := sq.Select("kv_val").Distinct().
+		From(o.TableName).
+		Where(sq.Eq{"kv_key": key}).
+		PlaceholderFormat(o.Placeholder)
+
+	query_text, args, err := query.ToSql()
```
Comment (thomas-fossati) [inline kvstore/sql.go:131]: nit: https://go.dev/doc/effective_go#mixed-caps

```suggestion
	queryText, args, err := query.ToSql()
```

File: kvstore/sql.go [hunk 2]
```diff
@@ -109,18 +123,24 @@ func (o SQL) Get(key string) ([]string, error) {
...
-	rows, err := o.DB.Query(q, key)
+	rows, err := o.DB.Query(query_text, args...)
 	if err != nil {
 		return nil, err
 	}
 
 	defer rows.Close()
 
-	var vals []string
+	var kv_val []string
```
Comment (thomas-fossati) [inline kvstore/sql.go:143]: ditto

File: kvstore/sql.go [hunk 3]
```diff
@@ -189,11 +211,15 @@ func (o SQL) Add(key string, val string) error {
...
-	// nolint:gosec
-	// o.TableName has been checked by isSafeTblName on init
-	q := fmt.Sprintf("INSERT INTO %s(key, vals) VALUES(?, ?)", o.TableName)
+	query := sq.Insert(o.TableName).Columns("kv_key", "kv_val").
+			Values(key, val).PlaceholderFormat(o.Placeholder)
 
-	_, err := o.DB.Exec(q, key, val)
+	query_text, args, err := query.ToSql()
+	if err != nil {
+		panic(err)
```
Comment (thomas-fossati) [inline kvstore/sql.go:219]: Any reason to `panic()`?

File: kvstore/sql.go [hunk 4]
```diff
@@ -217,19 +243,29 @@ func (o SQL) Set(key string, val string) error {
...
-	// nolint:gosec
-	// o.TableName has been checked by isSafeTblName on init
-	delQ := fmt.Sprintf("DELETE FROM %s WHERE key = ?", o.TableName)
+	delQuery := sq.Delete(o.TableName).
+			Where(sq.Eq{"kv_key": key}).
+			PlaceholderFormat(o.Placeholder)
+
+	query_text, args, err := delQuery.ToSql()
+	if err != nil {
+		panic(err)
```
Comment (thomas-fossati) [inline kvstore/sql.go:252]: ditto (`panic`ing in a package)

**Review by thomas-fossati — COMMENTED** [review]
LGTM, thanks.

A couple of nits and questions inlined.

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!

**Comment by yogeshbdeshpande** [comment]
I do not have additional comments, I looked at the code and I had same questions as asked by Thomas so once you address them, I am fine with the change!

File: kvstore/sql.go (hunk 3 above)
Comment (setrofim) [reply kvstore/sql.go:219]: The error only currs if the query is malfomed this something that needs to be fixed entirely in this file.

File: kvstore/sql.go (hunk 4 above)
Comment (setrofim) [reply kvstore/sql.go:252]: same reason as above -- this is a run-time check for a compile-time error.

File: kvstore/sql.go (hunk 3 above)
Comment (thomas-fossati) [reply kvstore/sql.go:219]: It makes sense if that's the only reason - apart from OOM, of course, which would kill the runtime anyway.

**Pull request merged**